### PR TITLE
Dropdown: refactor styles for Panel to use subComponentStyles

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [X] Addresses an existing issue: #5689
- [X] Include a change request file using `$ npm run change`

#### Description of changes

Refactors the styles for `Panel` in `Dropdown` to use `subComponentStyles`. Also, repurposing the issue it fixes for a different issue which had comments pointing to it.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts [hunk 1]
```diff
@@ -277,14 +263,24 @@ export const getStyles: IStyleFunction<IDropdownStyleProps, IDropdownStyles> = p
...
-    subComponentStyles: { label: { root: { display: 'inline-block' } } }
+    subComponentStyles: {
+      label: { root: { display: 'inline-block' } },
+      panel: {
+        root: [panelClassName],
+        main: {
+          // Force drop shadow even under medium breakpoint
+          boxShadow: '-30px 0px 30px -30px rgba(0,0,0,0.2)'
+        },
+        contentInner: { padding: '0 0 40px' }
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts]: Guessing the change from 20px to 40px was intentional?

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts [hunk 2]
```diff
@@ -231,13 +232,14 @@ export interface IDropdownStyles {
   /** Refers to the callout that hosts Dropdown options in larger viewports. */
   callout: IStyle;
 
-  /** Subcomponent styles. */
+  /** SubComponent styles. */
```
Comment (ecraig12345) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts]: This should probably either be left as "Subcomponent" or changed to "Sub-component"

**Comment by KevinTCoughlin** [comment]
Kudos for doing this @Vitalius1!

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts (hunk 1 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.styles.ts]: Nope... was doing some testing and forgot to change it back. Good catch 👍

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts (hunk 2 above)
Comment (Vitalius1) [reply packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.types.ts]: 👍 will get a fix. I actually was unsure about this change)))

**Comment by Vitalius1** [comment]
@JasonGore can I have your attention with this PR? Thanks!

**Pull request merged**